util/src/main/java/com/thoughtworks/go/util/command/StreamPumper.java

```diff
@@ -89,10 +94,9 @@ public void readToEnd() {
         }
     }
 
-    public static StreamPumper pump(InputStream stream, StreamConsumer streamConsumer, String prefix, String encoding) {
-        StreamPumper pumper = new StreamPumper(stream, streamConsumer, prefix, encoding);
-        new Thread(pumper).start();
-        return pumper;
+    public Future<Boolean> pump() {
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        return executorService.submit(this);
```

Comment:
Who shuts this down? The earlier thread would shutdown after the `StreamPumper` runnable returned from the `run` method.